Calculator/Calculator/Model/LinkedList.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  LinkedList.swift
+//  Calculator
+//
+//  Created by Daehoon Lee on 2023/05/31.
+//
+
+struct LinkedList<T> {
+    var head: Node<T>?
+    var tail: Node<T>?
+    
+    init(head: Node<T>?) {
+        self.head = head
+    }
+    
+    mutating func append(_ data: T) {
+        guard head != nil
+        else {
+            head = Node(data: data)
+            tail = head
+            return
+        }
+        
+        tail?.updateNext(Node(data: data))
+        tail = tail?.fetchNext()
+    }
+    
+    mutating func removeFirst() -> T? {
+        guard head != nil else { return nil }
+        
+        if head === tail { tail = nil }
```

Comment:
[If](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/controlflow#If)와 [Early Exit](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/controlflow#Early-Exit) 문서를 보면 기본적으로 코드 블록을 시작할 때 개행을 하는 것이 일반적일 것 같습니다.

Reply:
- 말씀하신 내용은 29번 줄과 31번 줄의 순서를 바꾸라는 의미일까요? 건디가 주신 링크의 자료를 봤지만 위의 내용에 대한 부분이 어떤 문장에서 나타나는 걸까요?
- 위의 순서를 바꾼다면 guard에서의 early exit의 의미가 조금은 약해진다고 느껴지는데 convention을 우선하는게 좋을까요?

Reply:
이 부분은 위의 코멘트와 마찬가지로 일관적으로 적용하면 문제없을 것 같습니다. 기본 형식은 위의 링크된 [Statements](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/statements)나 [A Swift Tour](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/guidedtour)를 참고하시면 될 것 같습니다. 

제가 말한 것은 아래와 같은 방식이었습니다.

```suggestion
        guard head != nil else {
            return nil
        }
        
        if head === tail { 
            tail = nil 
        }
```